tools/cli/go-whisk-cli/commands/action.go

```diff
@@ -404,41 +404,23 @@ func parseAction(cmd *cobra.Command, args []string, update bool) (*whisk.Action,
             }
         }
 
-        if flags.action.kind == "swift:3" || flags.action.kind == "swift:3.0" || flags.action.kind == "swift:3.0.0" {
-            action.Exec.Kind = "swift:3"
-        } else if flags.action.kind == "nodejs:6" || flags.action.kind == "nodejs:6.0" ||
-                flags.action.kind == "nodejs:6.0.0" {
-            action.Exec.Kind = "nodejs:6"
-        } else if flags.action.kind == "nodejs:default" {
-            action.Exec.Kind = "nodejs:default"
-        } else if flags.action.kind == "swift:default" {
-            action.Exec.Kind = "swift:default"
-        } else if flags.action.kind == "nodejs" {
-            action.Exec.Kind = "nodejs"
-        } else if flags.action.kind == "python" {
-            action.Exec.Kind = "python"
+        if len(flags.action.kind) > 0 {
```

Comment:
Just to be safe, you may also want to pass `flags.action.kind`, and `flags.action.docker` to the new method since those are globals and could be accidentally be overwritten.